src/vulkan/mod.rs

```diff
@@ -15,6 +15,19 @@ use crate::{
     allocator::fmt_bytes, AllocationError, AllocatorDebugSettings, MemoryLocation, Result,
 };
 
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum AllocationScheme {
+    /// This allocation will be for a buffer, and it will be driver managed using vulkans DedicatedAllocation feature.
+    /// The driver will be able to perform optimizations in some cases with this type of allocation.
+    DedicatedBuffer { buffer: vk::Buffer },
+    /// This allocation will be for a image, and it will be driver managed using vulkans DedicatedAllocation feature.
+    /// The driver will be able to perform optimizations in some cases with this type of allocation.
+    DedicatedImage { image: vk::Image },
+    /// This allocation will be managed by the GPU allocator.
+    /// It is possible that the allocation will reside in an allocation block shared with other resources.
+    GpuAllocatorManaged,
+}
+
```

Comment:
@MarijnS95 What do you think about this for the user exposed interface? 
We can now also correctly pass the buffer and image when required for dedicated allocations. I got rid of the `BlockType` as that was essentially serving the same purpose, just with less configurability and under the hood.

Reply:
Interface looks good, though I'd have reworded some of the comments:

```suggestion
#[derive(Copy, Clone, Debug, Eq, PartialEq)]
pub enum AllocationScheme {
    /// Perform a dedicated, driver-managed allocation for the given buffer, allowing
    /// it to perform optimizations on this type of allocation.
    DedicatedBuffer { buffer: vk::Buffer },
    /// Perform a dedicated, driver-managed allocation for the given image, allowing
    /// it to perform optimizations on this type of allocation.
    DedicatedImage { image: vk::Image },
    /// The memory for this resource will be allocated and managed by gpu-allocator.
    /// It is possible that the allocation will reside in [`vk::Memory`] shared with other resources.
    GpuAllocatorManaged,
}

```